src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -309,16 +321,13 @@ public override async Task<BaseConfiguration> GetBaseConfigurationAsync(Cancella
         /// </summary>
         public override void RequestRefresh()
```

Comment:
are there any notable changes in perf with this change?

Reply:
Somewhat not straight forward. Yes, but the behavior was wrong before. Before this and the last change, RequestRefresh moved to a background thread, so there was no immediate impact on performance of the calling thread to queue a refresh. This was incorrect because our code and others write to the contract that if a RequestRefresh is called, and enough time has passed, new configuration will be returned by GetConfigurationAsync. If it is on a background thread, we lose that guarantee.